### PR TITLE
chore: Added missing sources files in CMake

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented
Added missing sources files (index, interfaces, node, logging, util) in CMake so they can be indexed by IDE.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by UdjinM6** [comment]
Looks good but what about other missing folders like `index`, `interfaces` etc.?

**Comment by UdjinM6** [comment]
Looks better :) there is also `src/support/`, `src/test/util/` and `src/test/fuzz/`

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,6 +94,8 @@ file(GLOB SOURCE_FILES
         src/univalue/include/*.h
         src/univalue/lib/*.cpp
         src/univalue/lib/*.h
+        src/util/*.h
+        src/util/*.cpp
         src/wallet/*.cpp
         src/wallet/*.h
         src/wallet/test/*.cpp
```
Comment (knst) [inline CMakeLists.txt:114]: `src/wallet/test/*.h` due to `./wallet/test/wallet_test_fixture.h`

Also missing:
```
src/bench/data/*.h                                                              
src/config/*.h
src/logging/*.h
src/node/*.cpp
src/node/*.h
src/obj/*.h
src/support/allocators/*.h
src/support/*.cpp
src/support/*.h
src/test/fuzz/*.cpp
src/test/fuzz/*.h
src/test/util/*.cpp
src/test/util/*.h
src/univalue/gen/*.cpp
src/univalue/test/*.cpp
src/util/*.cpp
src/util/*.h
```

Should we include 3rd party libraries?
If so, missing also:
```
src/crc32c/include/crc32c/*.h
src/crc32c/src/*.h
src/crypto/ctaes/*.h
src/dashbls/src/*.cpp                                                           
```

I used this script to gather filelist:
```
find src/ -name '*.h' -or -name '*.cpp'    | sed -r 's/[^/]*.(h|cpp)$/*.\1/'  | sort | uniq > list.txt
```

**Review by knst — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by UdjinM6** [comment]
@knst 
```
src/bench/data/*.h -- autogenerated
src/config/*.h -- autogenerated
src/obj/*.h -- autogenerated
src/support/allocators/*.h -- GOOD CATCH!
src/univalue/gen/*.cpp -- I think we avoid internals
src/univalue/test/*.cpp -- I think we avoid internals
```

File: CMakeLists.txt (hunk 1 above)
Comment (knst) [reply CMakeLists.txt:114]: Ooops, I am little late.
Updated comment below.

Btw, there's still missing files:
```
src/univalue/test/*.cpp                                                         
src/wallet/test/*.h
src/univalue/gen/*.cpp
```

(same script)

should we include 3rd party libraries also?
```
src/crc32c/include/crc32c/*.h
src/crc32c/src/*.h
src/crypto/ctaes/*.h
src/dashbls/src/*.cpp  
src/secp256k1/include/*.h

**Comment by knst** [comment]
Even some files are autogenerated, better to have them in IDE! @ogabrielides I think these should be included also...
```
src/bench/data/*.h -- autogenerated
src/config/*.h -- autogenerated
src/obj/*.h -- autogenerated
```
@UdjinM6

**Comment by UdjinM6** [comment]
I don't use cmake or IDE that depends on `CMakeLists.txt` so I have no strong opinion on this but `src/bench/data/*.h` and `src/obj/*.h` look pretty useless imo 🤷‍♂️

**Comment by knst** [comment]
>  src/obj/*.h look pretty useless imo

src/obj/*.h has one define: `BUILD_GIT_COMMIT` that is used in `src/clientversion.cpp` 

It's indeed only one define, so, until you will be  curious whatis BUILD_GIT_COMMIT in clientversion.cpp this object file is indeed useless for code navigation/autocomplete.

Same for `src/bench/data/*.h` - it's only one symbol `block813851_raw` - until you don't need that - you don't need this file in CMakeLists.txt

But even if these 2 files cover very few rare cases it reduce diff between my script and current CMakeLists.txt -> next update of CMakeLists.txt would be easier because no need to re-decide fate of src/bench/data/*.h and src/obj/*.h - they are already in list.

**Comment by UdjinM6** [comment]
> But even if these 2 files cover very few rare cases it reduce diff between my script and current CMakeLists.txt -> next update of CMakeLists.txt would be easier because no need to re-decide fate of `src/bench/data/*.h` and `src/obj/*.h` - they are already in list.

makes sense I guess... or you could simply tweak the script to exclude them
```
find src/ -name '*.h' -or -name '*.cpp' | sed -r 's/[^/]*.(h|cpp)$/*.\1/' | sort | uniq | grep -Ev "/(bench/data|obj)/" > list.txt
```

I'm fine either way

**Comment by knst** [comment]
> or you could simply tweak the script to exclude them

Firstly, this script is not part of repo but my personal one, so, out-of-scope :)

Secondly, I didn't have this script before I have seen this repo and I won't safe it anywhere - easier to re-write that later, but I will loose this knowledge (that these files should be included)...

I vote either add this script to repo (somewhere to `contrib/devtools` either add auto-generated files to CMakeLists.txt.

**Comment by ogabrielides** [comment]
Let's wait for @PastaPastaPasta's opinion then

File: CMakeLists.txt (hunk 1 above)
Comment (PastaPastaPasta) [reply CMakeLists.txt:115]: I think it's good to add 3rd party libs as well

**Comment by knst** [comment]
> maybe add it as comment to the CMakeList.txt

I think this option is good enough because unlikely anyone will remember about a script in `contrib/devtools` which was my only concern.

**Review by knst — APPROVED** [review]
LGTM

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**